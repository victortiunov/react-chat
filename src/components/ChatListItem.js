import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withStyles } from 'material-ui/styles';
import { ListItem, ListItemText, ListItemIcon } from 'material-ui/List';
import Icon from 'material-ui-icons/Mail';
import Avatar from './Avatar';

const styles = () => ({
  activeChat: {
    backgroundColor: 'lightcyan',
  },
  newMessage: {
    color: '#64B5F6',
  },
});

class ChatListItem extends React.Component {
  static propTypes = {
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
    title: PropTypes.string.isRequired,
    disabled: PropTypes.bool.isRequired,
    isActive: PropTypes.bool.isRequired,
    unreadMessages: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
  };

  handleClick = () => {
    this.props.onClick();
  };

  render() {
    const title = this.props.title ? this.props.title : 'No Title';
    const {
      classes, disabled, isActive, unreadMessages,
    } = this.props;

    return (
      <ListItem
        divider
        className={classnames(isActive && classes.activeChat)}
        style={{ cursor: 'pointer' }}
        onClick={this.handleClick}
        disabled={disabled}
      >
        <Avatar colorFrom={title}>{title}</Avatar>
        <ListItemText primary={title} />
        {unreadMessages && (
          <ListItemIcon>
            <Icon className={classes.newMessage} />
          </ListItemIcon>
        )}
      </ListItem>
    );
  }
}

export default withStyles(styles)(ChatListItem);
