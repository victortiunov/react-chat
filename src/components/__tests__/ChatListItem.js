/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatListItem from '../ChatListItem';

const mockProps = {
  title: 'Gryffindor Talks',
  disabled: false,
  isActive: false,
  onClick: jest.fn(),
};

describe('<ChatListItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatListItem {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatListItem {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled correctly', () => {
    const tree = renderer.create(<ChatListItem {...mockProps} disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders active correctly', () => {
    const tree = renderer.create(<ChatListItem {...mockProps} isActive />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
