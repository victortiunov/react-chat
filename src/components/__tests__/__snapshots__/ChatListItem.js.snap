// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatListItem /> renders active correctly 1`] = `
<li
  className="MuiListItem-root-48 MuiListItem-default-51 MuiListItem-gutters-55 MuiListItem-divider-54 ChatListItem-activeChat-46"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <div
    className="MuiAvatar-root-58 MuiAvatar-colorDefault-59"
    style={
      Object {
        "backgroundColor": "#8bc34a",
      }
    }
  >
    GT
  </div>
  <div
    className="MuiListItemText-root-61"
  >
    <h3
      className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
    >
      Gryffindor Talks
    </h3>
  </div>
</li>
`;

exports[`<ChatListItem /> renders correctly 1`] = `
<li
  className="MuiListItem-root-48 MuiListItem-default-51 MuiListItem-gutters-55 MuiListItem-divider-54"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <div
    className="MuiAvatar-root-58 MuiAvatar-colorDefault-59"
    style={
      Object {
        "backgroundColor": "#8bc34a",
      }
    }
  >
    GT
  </div>
  <div
    className="MuiListItemText-root-61"
  >
    <h3
      className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
    >
      Gryffindor Talks
    </h3>
  </div>
</li>
`;

exports[`<ChatListItem /> renders correctly with new messages 1`] = `
<li
  className="MuiListItem-root-48 MuiListItem-default-51 MuiListItem-gutters-55 MuiListItem-divider-54"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <div
    className="MuiAvatar-root-58 MuiAvatar-colorDefault-59"
    style={
      Object {
        "backgroundColor": "#8bc34a",
      }
    }
  >
    GT
  </div>
  <div
    className="MuiListItemText-root-61"
  >
    <h3
      className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
    >
      Gryffindor Talks
    </h3>
  </div>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-92 MuiListItemIcon-root-91 ChatListItem-newMessage-47"
    color={undefined}
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
    />
  </svg>
</li>
`;

exports[`<ChatListItem /> renders disabled correctly 1`] = `
<li
  className="MuiListItem-root-48 MuiListItem-default-51 MuiListItem-gutters-55 MuiListItem-divider-54 MuiListItem-disabled-53"
  disabled={true}
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <div
    className="MuiAvatar-root-58 MuiAvatar-colorDefault-59"
    style={
      Object {
        "backgroundColor": "#8bc34a",
      }
    }
  >
    GT
  </div>
  <div
    className="MuiListItemText-root-61"
  >
    <h3
      className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
    >
      Gryffindor Talks
    </h3>
  </div>
</li>
`;
